File: Assets/Scripts/Network/Response/ResponseMovement.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ResponseMovementEventArgs : ExtendedEventArgs
{
	public int user_id { get; set; } // The user_id of whom who sent the request
	public float move_x { get; set; } // The movement in x direction
	public float move_y { get; set; } // The movement in y direction
	public float move_z { get; set; } // The movement in z direction
	public float rotate_x { get; set; } 
	public float rotate_y { get; set; } 
	public float rotate_z { get; set; } 
	public float rotate_w { get; set; } 

	public ResponseMovementEventArgs()
	{
		event_id = Constants.SMSG_MOVEMENT;
	}
}

public class ResponseMovement : NetworkResponse
{
	private int user_id;
	private float move_x;
	private float move_y;
	private float move_z;
	private float rotate_x;
	private float rotate_y;
	private float rotate_z;
	private float rotate_w;

	public ResponseMovement()
	{
	}

	public override void parse()
	{
		user_id = DataReader.ReadInt(dataStream);
		move_x = DataReader.ReadFloat(dataStream);
		move_y = DataReader.ReadFloat(dataStream);;
		move_z = DataReader.ReadFloat(dataStream);;
		rotate_x = DataReader.ReadFloat(dataStream);;
		rotate_y = DataReader.ReadFloat(dataStream);;
		rotate_z = DataReader.ReadFloat(dataStream);;
		rotate_w = DataReader.ReadFloat(dataStream);;
	}

	public override ExtendedEventArgs process()
	{
		ResponseMovementEventArgs args = new ResponseMovementEventArgs
		{
			user_id = user_id,
			move_x = move_x,
			move_y = move_y,
			move_z = move_z,
			rotate_x = rotate_x,
			rotate_y = rotate_y,
			rotate_z = rotate_z,
			rotate_w = rotate_w
		};

		return args;
	}
}

File: HW4_Server_Mk_2/src/networking/response/ResponseMovement.java
package networking.response;

// Other Imports
import metadata.Constants;
import model.Player;
import utility.GamePacket;
import utility.Log;
/**
 * The ResponseLogin class contains information about the authentication
 * process.
 */
public class ResponseMovement extends GameResponse {
    private Player player;
    private float move_x;
    private float move_y;
    private float move_z;
    private float rotate_x;
    private float rotate_y;
    private float rotate_z;
    private float rotate_w;

    public ResponseMovement() {
        responseCode = Constants.SMSG_MOVEMENT;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addInt32(player.getID());
        packet.addFloat(move_x);
        packet.addFloat(move_y);
        packet.addFloat(move_z);
        packet.addFloat(rotate_x);
        packet.addFloat(rotate_y);
        packet.addFloat(rotate_z);
        packet.addFloat(rotate_w);

        // Log.printf("In response movement, Player with id %d has moved to (%f, %f, %f)", player.getID(), move_y, move_y, move_z);
 
        return packet.getBytes();
    }

    public void setPlayer(Player player) {
        this.player = player;
    }

    public void setData(float move_x, float move_y, float move_z, float rotate_x, float rotate_y, float rotate_z, float rotate_w) {
        this.move_x = move_x;
        this.move_y = move_y; 
        this.move_z = move_z;
        this.rotate_x = rotate_x;
        this.rotate_y = rotate_y;
        this.rotate_z = rotate_z;
        this.rotate_w = rotate_w;
    }
}